Client/Application/AppDelegate.swift

```diff
@@ -78,6 +78,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
+        if let urlComponents = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) {
+            if urlComponents.scheme == "firefox" {
+                if urlComponents.host == "open-url" {
+                    if let query = urlComponents.query {
+                        if let newURLComponents = urlComponents.queryItems?.first as? NSURLQueryItem {
+                            if newURLComponents.name == "url" && newURLComponents.value != nil {
+                                if let newURL = NSURL(string: newURLComponents.value!) {
+                                    let tab = self.tabManager.addTab(request: NSURLRequest(URL: newURL))
+                                    self.tabManager.selectTab(tab)
+                                    return true
+                                }
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Great. This is pretty fail-safe I think. I'm fine with it, but if you want to simplify this `if let pyramid` .. since Swift 1.2 you can do this:

```
if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) where components.scheme == "firefox" && components.host == "open-url" {
    if let query = components.query, item =  components.queryItems?.first as? NSURLQueryItem {
        ... 
    }
}
```
So you can combine multiple `if let` conditionals. The expression will fail as a whole as soon as one check fails so it is safe to use previously unwrapped values. The `where` clause lets you do some basic conditional changes at the end.

I'm fine with how this looks now because it is pretty clear, but if you want to bring the number of lines down then you can use the above technique :-)